src/final2/subtests/InvalidCommandLineArgumentsTest.java

```diff
@@ -20,76 +20,105 @@ public InvalidCommandLineArgumentsTest() {
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=-270-90-315-45-90");
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=--270-90-315-45-90");
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=-270-90--315-45-90");
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule={270,90,315,45,90}");
-        
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-9999999999999999999999999999999999999-45-90");
-        
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "Rule=90-315-45-90-270");
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "ruLe=90-315-45-90-270");
-    }
-    
-    /**
-     * Asserts that the tested class prints an error message if an invalid speedup command is passed.
-     */
-    @Test
-    public void invalidSpeedupArgumentsTest() {
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "speedup=0");
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "speedup=1.2");
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "speedup=-1");
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "speedup=-5");
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "speedup=99999999999999999999999999999999999999999");
-        
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "speedUp=12");
-        errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "Speedup=12");
-    }
+	 * Asserts that the tested class prints an error message if an invalid rule command is passed.
+	 */
+	@Test
+	public void invalidRuleArgumentsTest() {
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-315-45-90-270");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-315-45");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-320-45-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=135-90-315-45-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-315-45.5-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90--315-45-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-315-45-90-");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-315-45-90--");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-90-315-45-90---");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=-270-90-315-45-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=--270-90-315-45-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=-270-90--315-45-90");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule={270,90,315,45,90}");
+		errorTest("quit", Input.getFile(TASK_SHEET_INPUT_FILE_1), "rule=270-0-315-45-90");
```

Comment:
These two lines are the ones I actually added. But due to formatting, `git` does not recognize that.